### PR TITLE
[Feature] Publish artifacts to nuget via appveyor

### PR DESCRIPTION
**Pull request by robertcoltheart**
This publishes `.nupkg` artifacts in appveyor, and allows deployment to Nuget when creating a release in GitHub. 

@jonnii need your help with the Nuget API key info. Must be encrypted using: https://ci.appveyor.com/tools/encrypt

### CODE REVIEW

**Comment by robertcoltheart** [comment]
P.S. This needs more work yet, don't merge.

**Comment by robertcoltheart** [comment]
Ok @jonnii take a 👀

File: .appveyor.yml [hunk 1]
```diff
@@ -2,13 +2,30 @@ image: Visual Studio 2017
...
+  - ps: .\build\run-tests.ps1
+
+artifacts:
+  - path: targets\**\*.nupkg
+
+deploy:
+  provider: NuGet
+  skip_symbols: true
+  api_key:
+    secure: token_etc
```
Comment (robertcoltheart) [inline .appveyor.yml]: Secure key goes here @jonnii

File: .appveyor.yml [hunk 2]
```diff
@@ -2,13 +2,30 @@ image: Visual Studio 2017
...
   project: src\Chinchilla.sln
-  
-test:
-  assemblies:
-    only:
-      - 'targets\specifications\**\*.Specifications.dll'
+  publish_nuget: true
+
+test_script:
+  - ps: .\build\run-tests.ps1
```
Comment (robertcoltheart) [inline .appveyor.yml]: `dotnet test` is not a first-class citizen of appveyor yet, which is why we need this PS script.

**Comment by robertcoltheart** [comment]
Actually a couple of things to change, like getting the proper version number. Will fix before this can be merged.

**Comment by robertcoltheart** [comment]
Ok, take a look. Versioning is done via `appveyor.yml` now.

**Comment by jonnii** [comment]
@robertcoltheart if you create a nugget account i'll give you push rights.

**Comment by robertcoltheart** [comment]
No, this won't push on every build. It will push to nuget when you create a release under Github. Username on Nuget is same as here on GH.

**Comment by robertcoltheart** [comment]
@jonnii take a look?

**Comment by jonnii** [comment]
@robertcoltheart it's all good, all you need is a nuget account and i'll give you push rights.

**Comment by jonnii** [comment]
@robertcoltheart i'll merge and we can figure the rest out after.

**Comment by robertcoltheart** [comment]
See: https://www.nuget.org/profiles/robertcoltheart

**Comment by jonnii** [comment]
@robertcoltheart donezo-

**Pull request merged**